lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -128,7 +165,76 @@ impl MessageState {
+#[derive(Debug, PartialEq, Eq)]
+pub enum ChunkedGelfDecompression {
+    Gzip,
+    Zlib,
+    None,
+}
+
+impl ChunkedGelfDecompression {
+    // TODO: add tests for compression detection
+    pub fn from_magic(data: &Bytes) -> Self {
+        if data.starts_with(GZIP_MAGIC) {
+            debug!("Detected Gzip compression");
+            return Self::Gzip;
+        }
+
+        if data.starts_with(ZLIB_MAGIC) {
+            if let Some([first_byte, second_byte]) = data.get(0..2) {
+                // TODO: add inspired from https://github.com/grafana/go-gelf/blob/25db8704bcf3f484c958312cd0cc49e5c768dcf1/gelf/reader.go#L120
+                // and https://github.com/Graylog2/graylog2-server/blob/1bdeca5f8ad3d31ff0c1c7981978ae8c4d300ebc/graylog2-server/src/main/java/org/graylog2/inputs/codecs/gelf/GELFMessage.java#L157
+                if (*first_byte as u16 * 256 + *second_byte as u16) % 31 == 0 {
+                    debug!("Detected Zlib compression");
+                    return Self::Zlib;
+                }
+            };
+            // TODO: should we return and error if the data starts with de ZLIB MAGIC
+            // but does not match the previous branches?
+            // for example, take a look to the graylog2server implementation
+            // https://github.com/Graylog2/graylog2-server/blob/1bdeca5f8ad3d31ff0c1c7981978ae8c4d300ebc/graylog2-server/src/main/java/org/graylog2/inputs/codecs/gelf/GELFMessage.java#L160
+        };
+
+        debug!("No compression detected",);
+        Self::None
+    }
+
+    pub fn decompress(&self, data: Bytes) -> Result<Bytes, ChunkedGelfDecompressionError> {
+        let decompressed = match self {
+            Self::Gzip => {
+                let mut decoder = MultiGzDecoder::new(data.reader());
+                let mut decompressed = Vec::new();
+                decoder
+                    .read_to_end(&mut decompressed)
+                    .context(GzipDecompressionSnafu)?;
+                Bytes::from(decompressed)
+            }
+            Self::Zlib => {
+                let mut decoder = ZlibDecoder::new(data.reader());
+                let mut decompressed = Vec::new();
+                decoder
+                    .read_to_end(&mut decompressed)
+                    .context(ZlibDecompressionSnafu)?;
+                Bytes::from(decompressed)
+            }
+            Self::None => data,
+        };
+        Ok(decompressed)
+    }
+}
+
+#[derive(Debug, Snafu)]
+pub enum ChunkedGelfDecompressionError {
+    #[snafu(display("Gzip decompression error: {source}"))]
+    GzipDecompression { source: std::io::Error },
+    #[snafu(display("Zlib decompression error: {source}"))]
+    ZlibDecompression { source: std::io::Error },
+}
+
+// TODO: add a comment that I removed the PartialEq and Eq derive due to
+// std::io::Error not implementing them. I had to refactor assert_eq to
+// assert!(matches!(..)) in tests due to this.
+#[derive(Debug, Snafu)]
```

Comment:
Removed derived `Eq` and `PartialEq` from this error, because `std::io::Erorr` did not implement them and they were required at `ChunkedGelfDecompressionError`